fl4health/parameter_exchange/sparse_coo_parameter_exchanger.py

```diff
@@ -0,0 +1,154 @@
+        """
+        Parameter exchanger for sparse tensors.
+
+        This exchanger is responsible for selecting an arbitrary subset of a model's parameters
+        via some selection criterion and then packaging them into the COO sparse tensor format for exchanging.
+
+        Args:
+            sparsity_level (float): The level of sparsity. Must be between 0 and 1.
+            score_gen_function (Callable[..., Dict[str, Tensor]]): Function that is responsible for
+            generating a score for every parameter inside a model in order to facilitate parameter selection.
+
+            In most cases, this function takes as inputs a current model and an initial model,
+            and it returns a dictionary that maps the name of each of the current model's tensors to
+            another tensor which contains the parameter scores.
+        """
+        assert 0 < sparsity_level <= 1
+        self.set_sparsity_level(sparsity_level)
+        self.parameter_packer: SparseCooParameterPacker = SparseCooParameterPacker()
+        self.set_score_gen_function(score_gen_function)
+
+    def set_sparsity_level(self, sparsity_level: float) -> None:
+        """This method exists to allow for setting new sparsity levels throughout training."""
+        self.sparsity_level = sparsity_level
+
+    def set_score_gen_function(self, score_gen_function: Callable[..., Dict[str, Tensor]]) -> None:
+        self.score_gen_function = score_gen_function
+
+    def generate_parameter_scores(self, model: nn.Module, initial_model: Optional[nn.Module]) -> Dict[str, Tensor]:
+        """Calling the score generating function to produce parameter scores."""
+        return self.score_gen_function(model, initial_model)
+
+    def create_masks(self, model: nn.Module, initial_model: Optional[nn.Module]) -> Dict[str, Tensor]:
+        """
+        Produce masks for model's tensors.
+
+        More precisely, this method first leverage a score generating function
+        to generate scores for all parameters of model.
+
+        Next, these scores are used to produce masks.
+        A threshold is determined according to the desired sparsity level,
+        then parameters whose scores are less than this threshold are assigned a mask value of 0,
+        while parameters whose scores are greater than or equal to this threshold are
+        assigned a mask value of 1.
+
+        Args:
+            model (nn.Module): Current model.
+            initial_model (nn.Module): Initial model.
+
+        Returns:
+            Dict[str, Tensor]: Dictionary that maps the name of each tensor to its corresponding mask.
+        """
+        all_parameter_scores = self.generate_parameter_scores(model, initial_model)
+        all_scores = torch.cat([val.flatten() for _, val in all_parameter_scores.items()])
+        # Sorting all scores and determining the threshold.
+        sorted_scores, _ = torch.sort(all_scores, descending=True)
```

Comment:
Can we use torch.quantile to avoiding having to sort all the parameters in the model?

Reply:
I thought maybe under the hood, quantile could do some binning to make it more efficient than sort but I did some profiling and the performance gain seems to be negligible so I think you can keep as is